src/test/java/teammates/test/util/TestUrl.java

```diff
@@ -0,0 +1,16 @@
+package teammates.test.util;
+
+import teammates.common.util.AppUrl;
+import teammates.test.driver.TestProperties;
+
+/**
+ * An extension of {@link AppUrl}, differing in the base URL used for conversion.
+ * In this class, the base URL used is the value of test.app.url in test.properties.
+ */
+public class TestUrl extends AppUrl {
+
+    public TestUrl(String url) {
+        super(TestProperties.inst().TEAMMATES_URL, url);
+    }
+    
+}
```

Comment:
What I'm suggesting is the constructor that takes base Url can be an additional constructor that need not be used all the time. In typical cases, use the constructor you are using now. You need to call that constructor only from the factory method in test code only.